common/tcbinfo.c

```diff
@@ -509,6 +509,201 @@ oe_result_t oe_parse_tcb_info_json(
     return result;
 }
 
+OE_INLINE uint32_t ReadUint32(const uint8_t* p)
+{
+    return p[0] | (p[1] << 8) | (p[2] << 16) | (p[3] << 24);
+}
+
+OE_INLINE uint64_t ReadUint64(const uint8_t* p)
+{
+    uint64_t temp = 0;
+    uint64_t result = 0;
+    for (int i = 7; i >= 0; i--)
+    {
+        temp = p[i];
+        result = (result << 8) | temp;
+    }
+    return result;
+}
+/**
+ * type = qe_identity
+ * Schema:
+ * {
+ *     "version" : integer,
+ *    "issueDate" : string,
+ *    "nextDate" : string,
+ *    "miscselect" : string,
+ *    "miscselectMask" : string,
+ *    "attributes" : string,
+ *    "attributesMask" : string,
+ *    "mrsigner" : hex string,
+ *    "isvprodid" : integer,
+ *    "isvsvn" : integer,
+ * }
+ */
+static oe_result_t _read_qe_identity_info(
+    const uint8_t** itr,
+    const uint8_t* end,
+    oe_parsed_qe_identity_info_t* parsed_info)
+{
+    oe_result_t result = OE_TCB_INFO_PARSE_ERROR;
+    uint64_t value = 0;
+    const uint8_t* date_str = NULL;
+    size_t date_size = 0;
+    uint8_t four_bytes_buf[4];
+    uint8_t sixteen_bytes_buf[16];
+
+    parsed_info->info_start = *itr;
+    OE_CHECK(_read('{', itr, end));
+
+    OE_TRACE_INFO("Reading version\n");
+    OE_CHECK(_read_property_name_and_colon("version", itr, end));
+    OE_CHECK(_read_integer(itr, end, &value));
+    parsed_info->version = (uint32_t)value;
+    OE_CHECK(_read(',', itr, end));
+
+    OE_TRACE_INFO("Reading issueDate\n");
+    OE_CHECK(_read_property_name_and_colon("issueDate", itr, end));
+    OE_CHECK(_read_string(itr, end, &date_str, &date_size));
+    if (oe_datetime_from_string(
+            (const char*)date_str, date_size, &parsed_info->issue_date) !=
+        OE_OK)
+        OE_RAISE(OE_TCB_INFO_PARSE_ERROR);
+    OE_CHECK(_read(',', itr, end));
+
+    // nextUpdate is treated as an optional property.
```

Comment:
Is nextUpdate optional?
Note: We can remove the optionality in tcb info as well since Intel has updated the format on their end.

Reply:
According to the latest Provisioning Certification Service for Intel SGX API documentation Revision: 0.8-draft, this is a mandatory field

Reply:
Since Intel has updated the format on their end, we ought to not make nextUpdate optional anymore.

Reply:
Updated PR addressed this.